waltid-libraries/waltid-sdjwt/src/commonMain/kotlin/id/walt/sdjwt/SDJwtVC.kt

```diff
@@ -0,0 +1,115 @@
+      } &&
+      !vct.isNullOrEmpty().also {
+        if(!it) message = "$message, VC has no verifiable credential type property (vct)"
+      } &&
+      verifyHolderKeyBinding(jwtCryptoProvider, requiresHolderKeyBinding, audience, nonce).also {
+        if(!it) message = "$message, holder key binding could not be verified"
+      }
+    )
+  }
+
+  companion object {
+
+    fun parse(sdJwt: String) = SDJwtVC(SDJwt.parse(sdJwt))
+
+    /**
+     * Parse SD-JWT VC from a token string and verify it
+     * @return SD-JWT VC verification result, with parsed SD-JWT VC
+     * @throws Exception if SD-JWT VC cannot be parsed
+     */
+    fun verifyAndParse(sdJwtVC: String, jwtCryptoProvider: JWTCryptoProvider, requiresHolderKeyBinding: Boolean,
+                       audience: String? = null, nonce: String? = null): VCVerificationResult {
+      val parsedVC = parse(sdJwtVC)
+      return parsedVC.verifyVC(jwtCryptoProvider, requiresHolderKeyBinding, audience, nonce)
+    }
+
+    fun sign(
+      sdPayload: SDPayload,
+      jwtCryptoProvider: JWTCryptoProvider,
+      issuerDid: String,
+      holderDid: String,
+      issuerKeyId: String? = null,
+      vct: String, nbf: Long? = null, exp: Long? = null, status: String? = null,
+      /** Set additional options in the JWT header */
+      additionalJwtHeader: Map<String, Any> = emptyMap()
+    ): SDJwtVC = doSign(sdPayload, jwtCryptoProvider, issuerDid, buildJsonObject {
+      put("kid", holderDid)
+    }, issuerKeyId, vct, nbf, exp, status, additionalJwtHeader)
+
+    fun sign(
+      sdPayload: SDPayload,
+      jwtCryptoProvider: JWTCryptoProvider,
+      issuerDid: String,
+      holderKeyJWK: JsonObject,
+      issuerKeyId: String? = null,
+      vct: String, nbf: Long? = null, exp: Long? = null, status: String? = null,
+      /** Set additional options in the JWT header */
+      additionalJwtHeader: Map<String, Any> = emptyMap()
+    ): SDJwtVC = doSign(sdPayload, jwtCryptoProvider, issuerDid, buildJsonObject {
+      put("jwk", holderKeyJWK)
+    }, issuerKeyId, vct, nbf, exp, status, additionalJwtHeader)
+
+    private fun doSign(
+      sdPayload: SDPayload,
+      jwtCryptoProvider: JWTCryptoProvider,
+      issuerDid: String,
+      cnf: JsonObject,
+      issuerKeyId: String? = null,
+      vct: String, nbf: Long? = null, exp: Long? = null, status: String? = null,
+      /** Set additional options in the JWT header */
+      additionalJwtHeader: Map<String, Any> = emptyMap()
+    ): SDJwtVC {
+      val undisclosedPayload = sdPayload.undisclosedPayload.toMutableMap().apply {
+        put("iss", JsonPrimitive(issuerDid))
+        put("cnf", cnf)
+        put("vct", JsonPrimitive(vct))
+        nbf?.let { put("nbf", JsonPrimitive(it)) }
+        exp?.let { put("exp", JsonPrimitive(it)) }
+        status?.let { put("status", JsonPrimitive(it)) }
+      }.let { JsonObject(it) }
+
+      val sdPayload = SDPayload(undisclosedPayload, sdPayload.digestedDisclosures)
```

Comment:
a method parameter with similar name already exists and being shadowed by this, is it intended?